core/src/main/java/hudson/init/InitMilestone.java

```diff
@@ -43,7 +43,10 @@
  *  <li>PLUGINS_PREPARED
  *  <li>PLUGINS_STARTED
  *  <li>EXTENSIONS_AUGMENTED
+ *  <li>SYSTEM_CONFIG_LOADED</li>
+ *  <li>SYSTEM_CONFIG_ADAPTED</li>
```

Comment:
Maybe I'm missing something, where is it added to the lifecycle?

Reply:
I was wondering the same thing 🤔

Reply:
If I understand correctly the magic inside Jenkins, it's added in the same way as EXTENSION_AUGMENTED, which is never attained directly:

https://github.com/jenkinsci/jenkins/blob/30dbf728ec6cee0b3c50e13f609fa419fe1782b6/core/src/main/java/jenkins/model/Jenkins.java#L965-L970